snarkos/ledger/mod.rs

```diff
@@ -125,6 +118,20 @@ impl<N: Network> Ledger<N> {
         // Return the next block.
         Ok(next_block)
     }
+
+    /// Attempts to add the given block to the ledger.
+    pub(crate) async fn add_next_block(self: &Arc<Self>, next_block: &Block<N>) -> Result<()> {
```

Comment:
if we passed `Block<N>` as value here, we could probably avoid having to clone it for the purpose of moving it into the blocking task

Reply:
done in https://github.com/AleoHQ/snarkOS/pull/1884, which is a follow-up to this PR